crates/wasm-encoder/src/component.rs

```diff
@@ -104,3 +97,19 @@ impl Default for Component {
         Self::new()
     }
 }
+
+impl Encode for Component {
+    fn encode<S>(&self, sink: &mut S)
```

Comment:
This seems somewhat surprising to me in that I would naively expect `Encode for Component` to look the same as `Component::finish`. Or otherwise `Component` here is "top-level" so I'd expect it to produce the top-level encoding as opposed to the nested encoding. Not exactly a huge issue but wanted to point this out.

Reply:
Indeed, this might be a point of confusion. Open to suggestions as how best to make it less so.

Reply:
I don't particularly feel strong enough that this is the "one blocker" or anything like that. Personally I thought the previous solution of something like `struct NestedComponentSection<'a>(pub &'a Component)` would work well. It's an extra type or two for modules and components but it felt like the best way to communicate intent.

Reply:
As I also didn't feel too strongly about it, I've restored `ModuleSection` and `NestedComponentSection`.